src/compiler/glue-compiler.cc

```diff
@@ -347,7 +347,7 @@ hilti::Result<std::string> GlueCompiler::getNextEvtBlock(std::istream& in, int*
 }
 
 void GlueCompiler::preprocessEvtFile(hilti::rt::filesystem::path& path, std::istream& in, std::ostream& out) {
-    hilti::util::SourceCodePreprocessor pp({{"ZEEK_VERSION", _zeek_version}});
+    hilti::util::SourceCodePreprocessor pp({{"ZEEK_VERSION", *_zeek_version}});
```

Comment:
The comparisons of `optional<int>` and `int` below are well-defined, but here we'd trigger UB if `_zeek_version` is unset.

Let's also add an `assert(_zeek_version)`.

```suggestion
    assert(_zeek_version);
    hilti::util::SourceCodePreprocessor pp({{"ZEEK_VERSION", *_zeek_version}});
```

Reply:
I added it as a gate in `loadEvtFile()`, that ensures it for other helpers downstream like this one.